### DISCUSSION TITLE
Border for text with 2 colors?

### DISCUSSION THREAD

**Discussion by jaka1-newbie** [ImageMagick/ImageMagick#2688] [Help] [answered]
I have an image . And i need to add text with this border.
Is it possible to add it?
![0-2-1-580x387](https://user-images.githubusercontent.com/67605008/95326050-475e0400-08a2-11eb-8b32-80256bfc93fe.jpg)

**Comment by snibgo** [comment]
What version of IM, on what platform?

I don't understand your question.

Perhaps you have an image that you want to be a background, and create a smaller image of text composited in the centre of the background image.

**Comment by jaka1-newbie** [comment]
Sorry for my bad english.
Tru to explain what i need.
I need from first image make second(or similar).
![3](https://user-images.githubusercontent.com/67605008/95366923-62496c00-08d4-11eb-9246-352450e12d03.jpg)
![0-2-1-580x387](https://user-images.githubusercontent.com/67605008/95366934-65445c80-08d4-11eb-9694-794cde9d66bd.jpg)

**Reply by fmw42** [reply]
Please explain/show where you want to add text in this image? Or do you want to replace "TERAZZO PATTERN" with something else. If the latter, do you have an image that has no text in the place you want to add it?

What is your IM version and platform?

**Reply by jaka1-newbie** [reply]
![3](https://user-images.githubusercontent.com/67605008/95367797-7c377e80-08d5-11eb-9a72-c6f4caed89de.jpg)
I'm using Im v.6, win 7 x64
In first image add text Terrazzo Pattern with white bacground and black border with shadow. As result - second image

**Reply by fmw42** [reply]
Sorry, do you want to add text to this image -- the one that has no text -- so that it looks like the one that has the TERRAZO text?

**Reply by jaka1-newbie** [reply]
I want add text to image without text. 
Image with text as example - what i want to get as result.

**Reply by fmw42** [reply]
I do not understand. How can you add text to an image without text?  Do you have a second image that you want to put into the first image. If so, post the second mage that should be put into the first image.

**Reply by jaka1-newbie** [reply]
Text that need to be added "Terrazzo Pattern"

**Comment by jaka1-newbie** [comment]
Fred!
You didn't understand what i say...

**Comment by fmw42** [comment, marked as answer]
Here is one way. There might be more efficient methods.

Unix syntax:

```
magick background.jpg \
\( -size 1000x600 -background white -fill black -gravity center -font Ubuntu label:"TERRAZO\nPATTERN" \
-bordercolor white -border 20 -bordercolor black -border 20 -bordercolor white -border 50 \) \
\( +clone -fill "gray(15%)" -colorize 100 -virtual-pixel white -blur 0x40 -level 0x50% \) \
\( -clone 1 -shave 40x40 -clone 2 +swap -gravity center -compose over -composite \) \
-delete 1,2 \
-gravity center -compose over -composite result.jpg
```

![result](https://user-images.githubusercontent.com/13458218/95375107-08c14d00-0894-11eb-81b8-368a84751214.jpg)

For Windows, change the end of line \ markers to ^, remove the \ escapes on the parentheses to just ( ... ). If in a .bat file, double the % to %%

**Reply by jaka1-newbie** [reply]
Great! Thx! Testing in progress...

**Comment by jaka1-newbie** [comment]
This code:
convert "D:\3.jpg" ^
( -size 1000x600 -background white -fill black -gravity center -font IM-FELL-English-Roman label:"TERRAZO\nPATTERN" ^
-bordercolor white -border 20 -bordercolor black -border 20 -bordercolor white -border 50 ) ^
( +clone -fill "gray(15%)" -colorize 100 -virtual-pixel white -blur 0x40 -level 0x50% ) ^
( -clone 1 -shave 40x40 -clone 2 +swap -gravity center -compose over -composite ) ^
-delete 1,2 ^
-gravity center -compose over -composite "D:\result.jpg"



This err:
convert: unable to open image '^

(': Invalid argument @ error/blob.c/OpenBlob/3539.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/562.
convert: unable to open image '^

-bordercolor': Invalid argument @ error/blob.c/OpenBlob/3539.
convert: unable to open image '^

-bordercolor': Invalid argument @ error/blob.c/OpenBlob/3539.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/562.
convert: unable to open image 'white': No such file or directory @ error/blob.c/OpenBlob/3539.
convert: unable to open image 'white': No such file or directory @ error/blob.c/OpenBlob/3539.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/562.
convert: unable to parse expression `)' @ error/convert.c/ConvertImageCommand/587.

**Comment by jaka1-newbie** [comment]
I think my problem in parentheses (((

**Reply by fmw42** [reply]
Could be your OS requires some other syntax. I am not an expert on Windows. Also it may not be able to access files from another volume.

**Comment by snibgo** [comment]
@jaka1-newbie: Is your command in a BAT file? If it is then you must double every percent sign: %%

**Reply by jaka1-newbie** [reply]
Tried
